src/Build/Logging/BinaryLogger/ProjectImportsCollectorExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Logging
+{
+    internal static class ProjectImportCollectorExtensions
+    {
+        private static string importingProject = ResourceUtilities.GetResourceString("ImportingProject");
+        private static int importingProjectFilePathStart = importingProject.IndexOf('"') + 1;
+        private static string importingProjectPrefix = importingProject.Substring(0, importingProjectFilePathStart);
+
+        public static void IncludeSourceFiles(this ProjectImportsCollector projectImportsCollector, BuildEventArgs e)
+        {
+            if (e is ProjectStartedEventArgs)
+            {
+                var projectStarted = (ProjectStartedEventArgs)e;
+                projectImportsCollector.AddFile(projectStarted.ProjectFile);
+            }
+            else if (e is BuildMessageEventArgs)
```

Comment:
Right now the StartsWith call we have there is dwarfed by all the disk access happening during the build, so even if there is any slowdown at all I think it's OK to take the hit (since it only happens with /bl).

All that said, I am banking on @cdmihai adding better BuildEventContext for evaluation so I can detect off of that in the future. Or we could utilize the Sender string, which is now unused and is always "MSBuild". If we set the Sender = "Evaluation" and compare to that, that'd be cheaper.

I'm really hesitant to add a new EventArgs subtype because there are assumptions everywhere about the possible set of EventArgs like here:
https://source.dot.net/#Microsoft.Build/BackEnd/Components/Logging/EventSourceSink.cs,297
Who knows what switch default blocks we'll start hitting in other infrastructure (e.g. CloudBuild??)